packages/ra-core/src/form/sanitizeEmptyValues.ts

```diff
@@ -7,7 +7,7 @@
 export const sanitizeEmptyValues = (values: any, record: any = {}): any => {
     const sanitizedValues = {};
     Object.keys(values).forEach(key => {
-        if (values[key] === '') {
+        if (values[key] == null || values[key] === '') {
```

Comment:
I don't understand why you updated this function

Reply:
For react-admin inputs (such as TextInput), when the field is `undefined` from the dataProvider:
- if you touch the field (e.g. change it to `'foo'` then back to `''`)
- if your field has a defaultValue but you clear it
=> In both cases we will send `"field": null`, although the field was absent from what the dataProvider returned.

This is because **sanitizeEmptyValues** only triggers on `''` but not on `null`, however our components now return `null`.

IMO in this case **sanitizeEmptyValues** should still remove the values from what's been sent (hence my change of the implementation is needed).

I'll add a story to play with this to the branch in any case so that you can see for yourself :wink: